mysql-test/suite/plugins/t/server_audit_query_seqnum.test

```diff
@@ -0,0 +1,103 @@
+--source include/have_innodb.inc
+--source include/have_plugin_auth.inc
+--source include/not_embedded.inc
+
+if (!$SERVER_AUDIT_SO) {
+  skip No SERVER_AUDIT plugin;
+}
+
+# An unfortunate wait for check-testcase.inc to complete disconnect.
+let count_sessions= 1;
+source include/wait_until_count_sessions.inc;
+
+let $MYSQLD_DATADIR= `SELECT @@datadir`;
+let SEARCH_FILE= $MYSQLD_DATADIR/server_audit_seqnum.log;
+
+--echo #
+--echo # MDEV-37004: Audit Plugin not in correct order for AFTER trigger
+--echo #
+
+# Install the plugin and configure it exactly as in the JIRA ticket
+install plugin server_audit soname 'server_audit';
+
+set global server_audit_file_path='server_audit_seqnum.log';
+set global server_audit_output_type=file;
+set global server_audit_mode=0;
+set global server_audit_events='QUERY';
+set global server_audit_logging=on;
+
+--echo # Create test database and tables as in JIRA
+CREATE DATABASE IF NOT EXISTS test;
+USE test;
+
+--echo # Create tables exactly as in MDEV-37004
+CREATE TABLE source (
+  id bigint(20) NOT NULL AUTO_INCREMENT,
+  PRIMARY KEY (id)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_uca1400_ai_ci;
+
+CREATE TABLE dest (
+  id bigint(20) NOT NULL,
+  PRIMARY KEY (id)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_uca1400_ai_ci;
+
+CREATE TABLE dest_2 (
+  id bigint(20) NOT NULL,
+  PRIMARY KEY (id)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_uca1400_ai_ci;
+
+--echo # Create the trigger exactly as in MDEV-37004
+DELIMITER //;
+CREATE TRIGGER test_trigger 
+AFTER INSERT ON source 
+FOR EACH ROW 
+INSERT INTO dest (id) VALUES(NEW.id)//
+DELIMITER ;//
+
+--echo # Create the trigger exactly as in MDEV-37004
+DELIMITER //;
+CREATE TRIGGER test_trigger_2
+AFTER INSERT ON dest 
+FOR EACH ROW 
+INSERT INTO dest_2 (id) VALUES(NEW.id)//
+DELIMITER ;//
+
+--echo # Insert a row to trigger the AFTER trigger
+INSERT INTO source VALUES (NULL);
+
+--echo # Insert another row to see the pattern
+INSERT INTO source VALUES (NULL);
+
+--echo # Test with multi-row insert
+INSERT INTO source VALUES (NULL), (NULL);
+
+--echo # Test with multi-row insert within transaction
+SET TRANSACTION ISOLATION LEVEL REPEATABLE READ;
+BEGIN;
+INSERT INTO source VALUES (NULL), (NULL);
+COMMIT;
+
+--echo # Clean up
+DROP TABLE source, dest;
+DROP DATABASE test;
+
+--echo # Wait for audit events to be written
+--sleep 1
```

Comment:
Can use `include/search_pattern_in_file.inc` rather than a timing to wait for the last entry, after turning logging off first.